misc/xml-to-record-converter/src/main/java/io/ballerina/xmltorecordconverter/XMLToRecordConverter.java

```diff
@@ -225,12 +235,23 @@ private static List<Node> getRecordFieldsForXMLElement(Element xmlElement, boole
 
             if (xmlNode.getNodeType() == org.w3c.dom.Node.ELEMENT_NODE) {
                 Element xmlElementNode = (Element) xmlNode;
+                NamedNodeMap xmlAttributesMap = xmlElement.getAttributes();
+                boolean hasAttributeWithSameName = false;
+                for (int j = 0; j < xmlAttributesMap.getLength(); j++) {
+                    if(xmlAttributesMap.item(j).getNodeName().equals(xmlElementNode.getNodeName())) {
+                        hasAttributeWithSameName = true;
+                        break;
+                    }
+                }
+                if (hasAttributeWithSameName) {
+                    continue;
+                }
```

Comment:
AFAIU, `xmlElement` remains the same throughout the for loop. Hence, can we create a single hashset and only perform the contains check within the loop?

```suggestion
                if (attributeNames.contains(xmlElementNode.getNodeName())) {
                    continue;
                }
```
Add this logic outside the for loop.
``` java
NamedNodeMap xmlAttributesMap = xmlElement.getAttributes();
Set<String> attributeNames = new HashSet<>();
for (int j = 0; j < xmlAttributesMap.getLength(); j++) {
    attributeNames.add(xmlAttributesMap.item(j).getNodeName());
}
```

Reply:
+1